commit bf73c02636b55197e7c7ff870739c98cb200ef1a
Author: Volker Krause <vkrause@kde.org>
Date:   Fri Sep 4 15:45:29 2015 +0200

    Remove KDELibs4Support dependency.

commit ce575030415015c38d3e035b9bc507b9d4ae5a72
Author: Volker Krause <vkrause@kde.org>
Date:   Fri Sep 4 15:44:41 2015 +0200

    Port away from deprecated Qt methods.
    
    Those are not available once we remove the KDELibs4Support dependency.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d2beff4..c4af8c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,8 @@ ecm_setup_version(${KIMAP_LIB_VERSION} VARIABLE_PREFIX KIMAP
 
 ########### Find packages ###########
 find_package(KF5CoreAddons ${KF5_VERSION} CONFIG REQUIRED)
-find_package(KF5KDELibs4Support ${KF5_VERSION} CONFIG REQUIRED)
+find_package(KF5I18n ${KF5_VERSION} CONFIG REQUIRED)
+find_package(KF5KIO ${KF5_VERSION} CONFIG REQUIRED)
 
 find_package(KF5Mime ${KMIME_LIBS_VERSION} CONFIG REQUIRED)
 
diff --git a/autotests/kimaptest/sslserver.cpp b/autotests/kimaptest/sslserver.cpp
index b4bd7d0..98e83d5 100644
--- a/autotests/kimaptest/sslserver.cpp
+++ b/autotests/kimaptest/sslserver.cpp
@@ -77,7 +77,9 @@ void SslServer::incomingConnection(qintptr handle)
 
     QSslKey ssl_key(staticKey(), QSsl::Rsa);
     QSslCertificate ssl_cert(staticCert());
-    Q_ASSERT(ssl_cert.isValid());
+    Q_ASSERT(QDateTime::currentDateTime() >= ssl_cert.effectiveDate());
+    Q_ASSERT(QDateTime::currentDateTime() <= ssl_cert.expiryDate());
+    Q_ASSERT(!ssl_cert.isBlacklisted());
 
     socket->setPrivateKey(ssl_key);
     socket->setLocalCertificate(ssl_cert);
@@ -86,7 +88,7 @@ void SslServer::incomingConnection(qintptr handle)
     socket->ignoreSslErrors();
     connect(socket, SIGNAL(sslErrors(QList<QSslError>)), this, SLOT(sslErrors(QList<QSslError>)));
     connect(socket, SIGNAL(error(QAbstractSocket::SocketError)), this, SLOT(error(QAbstractSocket::SocketError)));
-    if (mProtocol != QSsl::TlsV1) {
+    if (mProtocol != QSsl::TlsV1_0) {
         socket->startServerEncryption();
     }
     addPendingConnection(socket);
diff --git a/autotests/kimaptest/sslserver.h b/autotests/kimaptest/sslserver.h
index 0ecd9fd..e714f38 100644
--- a/autotests/kimaptest/sslserver.h
+++ b/autotests/kimaptest/sslserver.h
@@ -29,7 +29,7 @@ public:
     SslServer(QSsl::SslProtocol);
     virtual void incomingConnection(qintptr handle) Q_DECL_OVERRIDE;
 
-private slots:
+private Q_SLOTS:
     void sslErrors(const QList<QSslError> &errors);
     void error(QAbstractSocket::SocketError);
 
diff --git a/src/appendjob.cpp b/src/appendjob.cpp
index 4d72903..8584c0e 100644
--- a/src/appendjob.cpp
+++ b/src/appendjob.cpp
@@ -124,7 +124,7 @@ void AppendJob::doStart()
 
     if (!d->internalDate.isNull()) {
         const QDateTime utcDateTime = d->internalDate.toUTC();
-        parameters += " \"" + utcDateTime.toString(QString::fromAscii("dd-MMM-yyyy hh:mm:ss")).toLatin1() + " +0000" + '\"';
+        parameters += " \"" + utcDateTime.toString(QStringLiteral("dd-MMM-yyyy hh:mm:ss")).toLatin1() + " +0000" + '\"';
     }
 
     parameters += " {" + QByteArray::number(d->content.size()) + '}';
